FirebaseInstallations/Source/Library/FIRInstallationsErrorUtil.m

```diff
@@ -25,63 +29,83 @@ void FIRInstallationsItemSetErrorToPointer(NSError *error, NSError **pointer) {
 @implementation FIRInstallationsErrorUtil
 
 + (NSError *)keyedArchiverErrorWithException:(NSException *)exception {
-  // TODO: Form a proper error
-  return [NSError errorWithDomain:@"NSKeyedArchiver" code:-1 userInfo:exception.userInfo];
+  NSString *failureReason = [NSString
+      stringWithFormat:@"NSKeyedArchiver exception with name: %@, reason: %@, userInfo: %@",
+                       exception.name, exception.reason, exception.userInfo];
+  return [self installationsErrorWithCode:FIRInstallationsErrorCodeUnknown
+                            failureReason:failureReason
+                          underlyingError:nil];
+}
+
++ (NSError *)keyedArchiverErrorWithError:(NSError *)error {
+  NSString *failureReason = [NSString stringWithFormat:@"NSKeyedArchiver error"];
```

Comment:
"NSKeyedArchiver threw an exception."

Reply:
Well, this method is used for the `NSKeyedArchiver` version that just returns an error and does not throw. I added a dot at the end and kept the string unchanged. Please, let me know if you see a better option for the string.